guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,398 @@
+    magnitude_stddev=0.2,
+    rate=0.5,
+)
+
+
+def apply_rand_augment(inputs):
+    inputs["images"] = rand_augment(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(
+    apply_rand_augment, num_parallel_calls=tf.data.AUTOTUNE
+)
+
+"""
+And finally lets inspect some of the results:
+"""
+
+visualize_dataset(train_dataset, title="After RandAugment")
+
+"""
+Try tweaking the magnitude settings to see a wider variety of results.
+"""
+
+"""
+## CutMix and MixUp: Generate High Quality Inter-class Examples
+
+
+`CutMix` and `MixUp` allow us to produce inter-class examples.  CutMix randomly cuts out
+portions of one image and places them over another, and MixUp interpolates the pixel
+values between two images.  Both of these prevent the model from overfitting the
+training distribution and improve the likelihood that the model can generalize to out of
+distribution examples.  Additionally, CutMix prevents your model from over-relying on
+any particular feature to perform its classifications.
+
+In this example, we will use `CutMix` and `MixUp` independently in a manually created
+preprocessing pipeline.  In most state of the art pipelines images are randomly
+augmented by either `CutMix`, `MixUp`, or neither.  The function below implements this
+in an equal 1/3 split.
+
+Note that our `cut_mix_and_mix_up` function is annotated with a `tf.function` to ensure
+optimal performance.
+"""
+cut_mix = keras_cv.layers.CutMix()
+mix_up = keras_cv.layers.MixUp()
+
+
+@tf.function
+def cut_mix_and_mix_up(samples):
+    choice = tf.random.uniform((), minval=0, maxval=1, dtype=tf.float32)
+    if choice < 1 / 3:
+        return cut_mix(samples, training=True)
+    elif choice < 2 / 3:
+        return mix_up(samples, training=True)
+    else:
+        return samples
+
+
+train_dataset = load_dataset().map(
+    cut_mix_and_mix_up, num_parallel_calls=tf.data.AUTOTUNE
+)
+
+visualize_dataset(train_dataset, title="After CutMix and MixUp")
+
+"""
+Great!  Looks like we have successfully added `CutMix` and `MixUp` to our preprocessing
+pipeline.
+"""
+
+"""
+## Customizing Your Augmentation Pipeline
+
+Perhaps you want to exclude an augmentation from `RandAugment`, or perhaps you want to
+include the `GridMask()` as an option alongside the default `RandAugment` augmentations.
+
+KerasCV allows you to construct production grade custom data augmentation pipelines using
+the `keras_cv.layers.RandomAugmentationPipeline` layer.  This class operates similarly to
+`RandAugment`; selecting a random layer to apply to each image `augmentations_per_image`
+times.  `RandAugment` can be thought of as a specific case of
+`RandomAugmentationPipeline`.  In fact, our `RandAugment` implementation inherits from
+`RandomAugmentationPipeline` internally.
+
+In this example, we will create a custom `RandomAugmentationPipeline` by removing
+`RandomRotation` layers from the standard `RandAugment` policy, and substitutex a
+`GridMask` layer in its place.
```

Comment:
Being able to write your own augmentation chain makes things so interesting and concise. Thanks for this one.

Reply:
A quick thought:

Instead of the current augmentation pipeline (built with `RandomAugmentationPipeline`), do you want to implement a minimal SimCLR augmentation pipeline with it? Totally understand if not.

Reply:
Let's actually do that in the SimCLR guide: https://keras.io/examples/vision/semisupervised_simclr/